interfaces/devdoc/file_util_linux.md

```diff
@@ -0,0 +1,67 @@
+# file_util_linux
+
+## Overview
+
+Linux implementation of the file operation functions.
+
+## Exposed API 
+
+```c
+// Most defines removed for space constraints
+#define INVALID_HANDLE_VALUE            NULL
+
+typedef void *HANDLE;
+typedef void *LPSECURITY_ATTRIBUTES;
+
+MOCKABLE_FUNCTION(, HANDLE, ebs_create_file, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+MOCKABLE_FUNCTION(, bool, ebs_close_handle, HANDLE, handle_input);
+```
+
+### ebs_create_file
+
+```c
+MOCKABLE_FUNCTION(, HANDLE, ebs_create_file, const char*, lpFileName, unsigned long, dwDesiredAccess, unsigned long, dwShareMode, LPSECURITY_ATTRIBUTES, lpSecurityAttributes, unsigned long, dwCreationDisposition, unsigned long, dwFlagsAndAttributes, HANDLE, hTemplateFile);
+```
+
+`ebs_create_file` implements the creating file function in Linux. Uses the `open` function in C for Linux.
+
+**SRS_FILE_UTIL_LINUX_09_001: [** If the `full_file_name` input is either empty or `NULL`, `ebs_create_file` shall return an `INVALID_HANDLE_VALUE`. **]**
+
+**SRS_FILE_UTIL_LINUX_09_002: [** `ebs_create_file` shall allocate memory for the file handle. ebs_create_file will succeed and return a non-`NULL` value.  **]**
```

Comment:
> shall allocate memory [](http://example.com/codeflow?start=51&length=22)

Do we need to allocate memory? "open" returns a file handle, which can be used for other file operations.

In close handle, we just call "close" on the handle.

Similarly, in win32, CreateFileA gives us a file handle that can be used with other API's.

Unless we need to store additional state with the handle, I would think we can just return the result of `open` as our HANDLE.

Reply:
If we do need to malloc here, then close_handle is missing a spec that says we free it.

Reply:
We need to allocate memory to allow for threadpools later on. Even though I haven't written it in the spec yet, the malloc is necessary for that.